rslint_core/src/groups/errors/valid_typeof.rs

```diff
@@ -0,0 +1,116 @@
+use crate::rule_prelude::*;
+use ast::{BinExpr, Expr, UnaryOp};
+use rslint_parser::{SyntaxText, TextRange};
+use SyntaxKind::*;
+
+declare_lint! {
+    /**
+    Enforces the to use valid string literals in a `typeof` comparison.
+
+    ## Invalid Code Examples
+    ```ignore
+    typeof foo === "strnig"
+    typeof foo == "undefimed"
+    typeof bar != "nunber"
+    typeof bar !== "fucntion"
+    ```
+    */
+    #[derive(Default)]
+    ValidTypeof,
+    errors,
+    "valid-typeof"
+}
```

Comment:
This should include the `requireStringLiterals` option that eslint has, simply add a `pub require_string_literals: bool`, add `#[serde(default)]`, and add doc comments on top of the `pub ...` field in the macro. Moreover, id like a small explanation of what it considers valid string literals and why using an invalid one is wrong in the docs if you dont mind.

Reply:
Ah okay. I didn't know how to add options 😅

Reply:
Yeah i realized i didnt talk about that in the dev docs, ill try to add a section for that soon